packages/mattermost-redux/src/actions/users.ts

```diff
@@ -104,113 +100,25 @@ export function loginById(id: string, password: string, mfaToken = ''): ActionFu
-        const state = getState();
-        const collapsedThreads = isCollapsedThreadsEnabled(state);
-        dispatch({
-            type: UserTypes.RECEIVED_ME,
-            data,
-        });
-
-        Client4.setUserId(data.id);
-        Client4.setUserRoles(data.roles);
-        let teamMembers;
-
-        try {
-            const membersRequest: Promise<TeamMembership[]> = Client4.getMyTeamMembers();
-            const unreadsRequest = Client4.getMyTeamUnreads(collapsedThreads);
-
-            teamMembers = await membersRequest;
-            const teamUnreads = await unreadsRequest;
-
-            if (teamUnreads) {
-                for (const u of teamUnreads) {
-                    const index = teamMembers.findIndex((m) => m.team_id === u.team_id);
-                    const member = teamMembers[index];
-                    member.mention_count = u.mention_count;
-                    member.msg_count = u.msg_count;
-                    member.mention_count_root = u.mention_count_root;
-                    member.msg_count_root = u.msg_count_root;
-                    if (collapsedThreads) {
-                        member.thread_count = u.thread_count;
-                        member.thread_mention_count = u.thread_mention_count;
-                    }
-                }
-            }
-        } catch (error) {
-            dispatch(batchActions([
-                {type: UserTypes.LOGIN_FAILURE, error},
-                logError(error),
-            ]));
-            return {error};
-        }
-
-        const promises = [
-            dispatch(getMyPreferences()),
-            dispatch(getMyTeams()),
-            dispatch(getClientConfig()),
-        ];
-
-        const serverVersion = Client4.getServerVersion();
-        dispatch(setServerVersion(serverVersion));
-
-        try {
-            await Promise.all(promises);
-        } catch (error) {
-            dispatch(batchActions([
-                {type: UserTypes.LOGIN_FAILURE, error},
-                logError(error),
+                logError(error as ServerError),
             ]));
             return {error};
         }
 
-        dispatch(batchActions([
-            {
-                type: TeamTypes.RECEIVED_MY_TEAM_MEMBERS,
-                data: teamMembers,
-            },
-            {
-                type: UserTypes.LOGIN_SUCCESS,
-            },
-        ]));
-        const roles = new Set<string>();
-        for (const role of data.roles.split(' ')) {
-            roles.add(role);
-        }
-        for (const teamMember of teamMembers) {
-            for (const role of teamMember.roles.split(' ')) {
-                roles.add(role);
-            }
-        }
-        if (roles.size > 0) {
-            dispatch(loadRolesIfNeeded(roles));
-        }
-
-        return {data: true};
+        return loadMe(true)(dispatch, getState);
     };
 }
 
-export function loadMe(): ActionFunc {
+export function loadMe(isLogginIn = false): ActionFunc {
```

Comment:
Nit: Should this be `isLoggingIn` ? I kinda like `isLogginIn` more, though :)

Reply:
oh boy 🤣

Reply:
✅  Updated as per request/discussion